lib/Runtime/Library/JavascriptLibrary.cpp

```diff
@@ -754,10 +754,15 @@ namespace Js
             }
         }
 
-        if(scriptFunction && (useAnonymous || scriptFunction->GetFunctionProxy()->EnsureDeserialized()->GetIsStaticNameFunction()))
+        if (scriptFunction)
         {
-            return true;
+            function->SetPropertyWithAttributes(PropertyIds::length, TaggedInt::ToVarUnchecked(function->GetFunctionProxy()->EnsureDeserialized()->GetReportedInParamsCount() - 1), PropertyConfigurable, nullptr, PropertyOperation_None, SideEffects_None);
+            if ((useAnonymous || scriptFunction->GetFunctionProxy()->EnsureDeserialized()->GetIsStaticNameFunction()))
```

Comment:
Cache `scriptFunction->GetFunctionProxy()->EnsureDeserialized()` somewhere? The compiler may be smart enough to cache this, but it would make these two lines look a bit nicer.